README.md

```diff
@@ -90,38 +94,57 @@ export default defineConfig([
     ```bash
     > ng g angular-eslint:add-eslint-to-project <PROJECT NAME>
     > ng config cli.schematicCollections "[\"angular-eslint\"]"
     ```
-3. Use the NI configured rules for Angular TypeScript code and Angular templates in the [ESLint flat configuration](https://eslint.org/docs/latest/use/configure/configuration-files-new) (`eslint.config.js`). Set the `parserOptions.project` configuration to the project's TypeScript configuration to correctly enable [linting with type information](https://typescript-eslint.io/getting-started/typed-linting)..
+4. Use the NI configured rules for Angular TypeScript code and Angular templates in the [ESLint flat configuration](https://eslint.org/docs/latest/use/configure/configuration-files-new) (`eslint.config.mjs`).
     ```js
     import { defineConfig } from 'eslint/config';
     import { angularTypescriptConfig, angularTemplateConfig } from '@ni/eslint-config-angular';
+    import { javascriptConfig } from '@ni/eslint-config-javascript';
 
     export default defineConfig([
+        {
+            // JavaScript rules fail to parse the HTML files that are added below. Therefore, the JavaScript
+            // configuration must now match the correct files to avoid an error.
+            files: ['**/*.js'],
+            extends: javascriptConfig
+        }, 
         {
             files: ['**/*.ts'],
             extends: angularTypescriptConfig,
             languageOptions: {
                 parserOptions: {
-                    project: ['./tsconfig.json'],
-                    tsConfigRootDir: import.meta.dirname,
-                },
-            },
+                    // The `languageOptions.parserOptions.projectService` typescript-eslint option does not identify
+                    // tsconfig.*.json files. Use the older `project` configuration instead. The general `tsconfig.json` is
+                    // defined last in the hope that the application and test configurations will apply to their
+                    // applicable files correctly.
+                    // https://typescript-eslint.io/troubleshooting/typed-linting/#project-service-issues
+                    project: ['tsconfig.app.json', 'tsconfig.spec.json', 'tsconfig.json']
+                }
+                // In projects (e.g. libraries) using `parserOptions.project`, Angular requires that the paths be
+                // relative to the root, but the VSCode extension requires them to be relative to the project
+                // directory. Set the root to be the project directory to satisfy both. The `parserOptions.projectService`
+                // configuration would likely resolve this, but is not used for reasons described above.
+                // https://typescript-eslint.io/blog/project-service
+                // tsconfigRootDir: import.meta.dirname
```

Comment:
Could we suggest always configuring it or is it required to be conditionally enabled for some reason?